### PR TITLE
feat: add nginx based docker compose

### PR DESCRIPTION
**Pull request by jainpawan21**
### What changed? Why was the change needed?

add nginx based reverse proxy setup with lets encrypt 

Docs PR:- https://github.com/novuhq/docs/pull/587
### Screenshots

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

### CODE REVIEW

File: docker/local/deployment/nginx-based/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+      options:
+        max-size: '50m'
+        max-file: '5'
+    expose:
+      - "${WS_PORT}"
+    environment:
+      VIRTUAL_HOST: ${REACT_APP_WS_HOST}
+      VIRTUAL_PORT: ${WS_PORT}
+      LETSENCRYPT_HOST: ${REACT_APP_WS_HOST}
+      LETSENCRYPT_EMAIL: ${LETSENCRYPT_EMAIL}
```
Comment (Cliftonz) [inline docker/local/deployment/nginx-based/docker-compose.yml:126]: How does the system leverage this env variable?

File: docker/local/deployment/nginx-based/.env.example [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+API_PORT=3000
+REDIS_PORT=6379
+REDIS_CACHE_SERVICE_PORT=6379
+WS_PORT=3002
+WEB_PORT=4200
+EMBED_PORT=4701
+WIDGET_PORT=4500
+
+# Hosts
+REACT_APP_WS_HOST=ws.novu.domain.com
```
Comment (LetItRock) [inline docker/local/deployment/nginx-based/.env.example:42]:
```suggestion
WS_HOST=ws.novu.domain.com
```

File: docker/local/deployment/nginx-based/.env.example [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+# General
+# available values 'dev', 'test', 'production', 'ci', 'local'
+NODE_ENV=production
+
+# MongoDB
+MONGO_MAX_POOL_SIZE=500
+MONGO_MIN_POOL_SIZE=100
+MONGO_INITDB_ROOT_USERNAME=novu-selfhost
+MONGO_INITDB_ROOT_PASSWORD=qwertyuiop
+MONGO_URL=mongodb://novu-selfhost:qwertyuiop@mongodb:27017/novu-db?authSource=admin
```
Comment (LetItRock) [inline docker/local/deployment/nginx-based/.env.example:18]:
```suggestion
MONGO_URL=mongodb://$MONGO_INITDB_ROOT_USERNAME:$MONGO_INITDB_ROOT_PASSWORD@mongodb:27017/novu-db?authSource=admin
```

File: docker/local/deployment/nginx-based/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+      - 27017:27017
+    restart: unless-stopped
+  
+  api:
+    image: 'ghcr.io/novuhq/novu/api:latest'
+    depends_on:
+      - mongodb
+      - redis
+      - nginx-proxy
+      - nginx-proxy-acme
```
Comment (LetItRock) [inline docker/local/deployment/nginx-based/docker-compose.yml:35]: `nginx-proxy-acme` depends on `nginx-proxy`, so we can just have `nginx-proxy-acme` here and in some other places as well

File: docker/local/deployment/nginx-based/docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+      API_ROOT_URL: ${API_ROOT_URL}
+      DISABLE_USER_REGISTRATION: ${DISABLE_USER_REGISTRATION}
+      PORT: ${API_PORT}
+      FRONT_BASE_URL: ${FRONT_BASE_URL}
+      MONGO_URL: ${MONGO_URL}
+      MONGO_MIN_POOL_SIZE: ${MONGO_MIN_POOL_SIZE}
+      MONGO_MAX_POOL_SIZE: ${MONGO_MAX_POOL_SIZE}
+      REDIS_HOST: ${REDIS_HOST}
+      REDIS_PORT: ${REDIS_PORT}
+      REDIS_DB_INDEX: 2
```
Comment (LetItRock) [inline docker/local/deployment/nginx-based/docker-compose.yml:60]: let's have this in the variable as well

File: docker/local/deployment/nginx-based/docker-compose.yml (hunk 1 above)
Comment (LetItRock) [reply docker/local/deployment/nginx-based/docker-compose.yml:126]: https://github.com/nginx-proxy/nginx-proxy

**Pull request closed without merging**